pkg/resources/configprovider/crud.go

```diff
@@ -0,0 +1,199 @@
+		return
+	}
+
+	res.Diagnostics.Append(res.State.Set(ctx, addonConfigProvider)...)
+	if res.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read resource information
+func (r *ResourceConfigProvider) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	tflog.Debug(ctx, "ConfigProvider READ", map[string]any{"request": req})
+
+	addonConfigProvider := helper.StateFrom[ConfigProvider](ctx, req.State, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	addonEnvRes := tmp.GetConfigProviderEnv(ctx, r.Client(), r.Organization(), addonConfigProvider.ID.ValueString())
+	if addonEnvRes.HasError() {
+		resp.Diagnostics.AddError("failed to get add-on env", addonEnvRes.Error().Error())
+		return
+	}
+
+	// Convert the environment variables to a map
+	envVars := map[string]string{}
+	for _, envVar := range *addonEnvRes.Payload() {
+		envVars[envVar.Name] = envVar.Value
+	}
+
+	// Update the environment in the state
+	addonConfigProvider.fromEnv(ctx, envVars, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Update the state
+	resp.Diagnostics.Append(resp.State.Set(ctx, addonConfigProvider)...)
+}
+
+// Update resource
+func (r *ResourceConfigProvider) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	plan := helper.PlanFrom[ConfigProvider](ctx, req.Plan, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	state := helper.StateFrom[ConfigProvider](ctx, req.State, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if plan.ID.ValueString() != state.ID.ValueString() {
+		resp.Diagnostics.AddError("configProvider cannot be updated", "mismatched IDs")
+		return
+	}
+
+	// Only name can be edited
+	addonRes := tmp.UpdateAddon(ctx, r.Client(), r.Organization(), plan.ID.ValueString(), map[string]string{
+		"name": plan.Name.ValueString(),
+	})
+	if addonRes.HasError() {
+		resp.Diagnostics.AddError("failed to update ConfigProvider", addonRes.Error().Error())
+		return
+	}
+
+	tflog.Debug(ctx, "Updating environment variables")
+	envVars := plan.toEnv(ctx, &resp.Diagnostics)
+
+	// Always initialize an empty slice, even if there are no environment variables
+	envVarsArray := []tmp.EnvVar{}
+
+	// Only add environment variables if the map is not empty
+	if len(envVars) > 0 {
+		// Convert the map to the expected format: []tmp.EnvVar
+		for k, v := range envVars {
+			envVarsArray = append(envVarsArray, tmp.EnvVar{
+				Name:  k,
+				Value: v,
+			})
+		}
+	}
+
+	tflog.Debug(ctx, "Environment variables to update", map[string]interface{}{"count": len(envVarsArray)})
+	envRes := tmp.UpdateConfigProviderEnv(ctx, r.Client(), r.Organization(), plan.ID.ValueString(), envVarsArray)
+	if envRes.HasError() {
+		resp.Diagnostics.AddError("failed to configure application environment", envRes.Error().Error())
+		return
+	}
+	state.Name = plan.Name
+	state.Environment = plan.Environment
```

Comment:
It works, but more accurate if you use the response of the API instead of the plan values